.travis.yml

```diff
@@ -16,6 +16,7 @@ before_cache:
   - rm -fv $HOME/.cabal/packages/hackage.haskell.org/01-index.tar.idx
 
   - rm -rfv $HOME/.cabal/packages/head.hackage
+  - cabal new-configure --enable-tests --disable-optimization --write-ghc-environment-files=always --jobs=2
```

Comment:
I don’t think I understand this.

1. Doesn’t `before_cache` get run on every build anyway? Why doesn’t that affect the _next_ build? Is it because we don’t cache the `cabal.project.local` file?

2. Given that this isn’t run until _after_ the build, and given that we don’t cache the `cabal.project.local` file, how does this result in us actually building the tests correctly at -O0 and all the rest?

I’m particularly concerned that if we change one of these flags in some important way, that change won’t be reflected in the builds.

Reply:
1. I think this comes down to us not caching the `cabal.project.local` file, yes.
2. N.B. we’re running the tests at `-O1` in Travis.

Reply:
> 2\. N.B. we’re running the tests at `-O1` in Travis.

How? Why do we use `--disable-optimization` at all, then? Was that the case on `master`?